numba/cuda/reduction.py

```diff
@@ -193,7 +239,8 @@ def copy_strides(arr, n, stride, tpb):
         while size >= 16:
             # Find the closest size that is power of two
             p2size = 2 ** int(math.log(size, 2))
-            # Plan for p2size
+            # Generate a plan for invoking kernels to reduce the first p2size
+            # elements
             plan = self._plan(p2size, nbtype, init)
```

Comment:
I'm still confused. If you unfold the code below, you'll see only `plan[0]` is used... So the multiple stages in the `plan` list don't seem useful at all?